### DISCUSSION TITLE
More tooling needed

### DISCUSSION THREAD

**Discussion by czabaj** [statelyai/xstate#1990] [General]
First and foremost, thank You for such a great implementation of FSM in JavaScript 👍 

I see great potential in front-end development. The FSM aligns nicely with wireframes for example. The visualization is super helpful for prototyping and designing but working in the xstate.viz lacks the comfort of IDE and IDE lacks the visualization capability. I can imagine workflow similar to Markdown plugins, where your editor window will be split into two sections, one with the code and one with FSM visualization.

Let's create a list of tools that are already available and sketch some must-have tools for eager developers to come up with.

I already mentioned a possible plugin for IDE, but I can also imagine some integration with wireframe apps, where the created wires could be exported as simple SCXML (or preferably JSON for XState) or FSM representation could be imported into the wireframe tool to create a template of screens and its interconnections.

**Comment by czabaj** [comment]
Not to mention, the ability to export SCXML from the diagram drawing apps would be also super helpful and if the app will be available on my tablet to draw with the pencil, I will be in heaven.

**Reply by davidkpiano** [reply]
This is exactly what I'm working on right now, and hopefully releasing within a month.

**Comment by Andarist** [comment]
There are plans for all of this but it takes times to implement it and we are doing it in our free time so please be patient. If u want to help out - please reach out to us on our Discord so we could discuss how you could help
